### ISSUE TITLE
Compile times of array-literal arguments scale with array size

### ISSUE THREAD

**Issue by elliottslaughter** [zdevito/terra#231]
`callFunc` in the following code takes a very long time to compile:

```
local WIDTH = 388
local HEIGHT = 185

terra func(arr2d : float[WIDTH][HEIGHT])
end
print("before compiling func")
func:compile()
print("after compiling func")

terra callFunc()
  var arr2d : float[WIDTH][HEIGHT]
  func(arr2d)
end
print("before compiling callFunc")
callFunc:compile()
print("before compiling callFunc")
```

It appears, at first pass, that the issue is the size of the array being passed. If you change `WIDTH` and `HEIGHT` to small numbers (e.g. 2), the code compiles instantly. Presumably, LLVM is doing something silly like exploding the array into its constituent elements and attempting to analyze them individually.

I realize that this may well be an LLVM bug rather than a Terra one, but as an end-user I generally expect compile-times for small functions like this to be well-behaved.

Here is an example stack trace from where things get stuck in LLVM:

```
#0  0x00000000014b16de in llvm::LoadAndStorePromoter::run(llvm::SmallVectorImpl<llvm::Instruction*> const&) const ()
#1  0x0000000001233825 in (anonymous namespace)::SROA::promoteAllocas(llvm::Function&) ()
#2  0x000000000123c8f7 in (anonymous namespace)::SROA::runOnFunction(llvm::Function&) ()
#3  0x0000000000e0f277 in llvm::FPPassManager::runOnFunction(llvm::Function&) ()
#4  0x0000000000e0f336 in llvm::legacy::FunctionPassManagerImpl::run(llvm::Function&) ()
#5  0x0000000000e0f414 in llvm::legacy::FunctionPassManager::run(llvm::Function&) ()
#6  0x00000000007b4309 in FunctionEmitter::emitFunction (this=0x7fffffffdff0, funcobj_=0x7fffffffdfe0)
    at src/tcompiler.cpp:1281
#7  0x00000000007a8369 in EmitFunction (CU=0x30ed930, funcdecl=0x7fffffffe170, user=0x0) at src/tcompiler.cpp:2551
#8  0x00000000007a867e in terra_compilationunitaddvalue (L=0x40000378) at src/tcompiler.cpp:2580
```

**Comment by elliottslaughter** [comment]
This is in LLVM 3.5.

**Comment by elliottslaughter** [comment]
Looks like this is already fixed in LLVM 3.8. Sorry about the noise.
